std/xml.d

```diff
@@ -562,7 +562,7 @@ class Document : Element
         string tagString = xml.tag.tagString;
 
         this(xml.tag);
-        prolog = s[0 .. tagString.ptr - s.ptr];
+        prolog = s[0 .. &tagString[0] - &s[0]];
```

Comment:
Huh... I'm not seeing why this is any safer than doing arithmetic on `.ptr`. It's still pointer arithmetic!